scipy/optimize/_differentialevolution.py

```diff
@@ -742,10 +748,21 @@ def _rand1(self, samples):
         return (self.population[r0] + self.scale *
                 (self.population[r1] - self.population[r2]))
 
-    def _randtobest1(self, candidate, samples):
+    def _randtobest1(self, samples):
         """
         randtobest1bin, randtobest1exp
         """
+        r0, r1, r2 = samples[:3]
+        bprime = np.copy(self.population[r0])
+        bprime += self.scale * (self.population[0] - bprime)
```

Comment:
Perhaps it's better to:
```
bprime = self.scale * (self.population[0] - bprime)
bprime += self.scale * (self.population[r1] - self.population[r2])
bprime += self.population[r0]
```
Avoiding the copy.

Reply:
Agreed. i was just keeping the same format as the original code.

It could be rearranged into a one-liner (spaced over multiple lines for clarity) too.